cmd/common/options.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import (
+	"os"
+
+	"github.com/djmitche/dd-agent-comp-experiments/comp/autodiscovery"
+	"github.com/djmitche/dd-agent-comp-experiments/comp/core"
+	"github.com/djmitche/dd-agent-comp-experiments/pkg/util/startup"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+)
+
+// SharedOptions defines fx.Options that are shared among all agent flavors.
+//
+// The confFilePath is passed to the comp/core/config component.
+//
+// If oneShot is true, then this is a "one-shot" process and all support for long-term
+// execution, such as health monitoring, will be disabled.
+func SharedOptions(confFilePath string, oneShot bool) fx.Option {
+	options := []fx.Option{}
+
+	var autoStart startup.AutoStart
+	if oneShot {
```

Comment:
I am wondering when `oneShot == true`, if it would be possible to adapt the`fx.Invoke` by having `fx.Invoke(RunAll)` and `fx.Invoke(RunOneShot)` with `RunOneShot(onShotDeps)` and `RunAll(alldeps)`

Reply:
Yeah, that might avoid the `[]fx.Option` appending.